internal/versionsapi/cli/add.go

```diff
@@ -0,0 +1,277 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/internal/logger"
+	"github.com/edgelesssys/constellation/v2/internal/versionsapi"
+	verclient "github.com/edgelesssys/constellation/v2/internal/versionsapi/client"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/mod/semver"
+)
+
+func newAddCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add a new version",
+		Long:  "Add a new version to the versions API.",
+		RunE:  runAdd,
+	}
+
+	cmd.Flags().String("ref", "", "Ref of the version to add")
+	cmd.Flags().String("stream", "", "Stream of the version to add")
+	cmd.Flags().String("version", "", "Version to add (format: \"v1.2.3\")")
+	cmd.Flags().Bool("latest", false, "Whether the version is the latest version of the ref/stream")
+	cmd.Flags().Bool("release", false, "Whether the version is a release version")
+	cmd.Flags().Bool("dryrun", false, "Whether to run in dry-run mode (no changes are made)")
+	cmd.Flags().Bool("debug", false, "Whether to enable debug logging")
```

Comment:
`debug` could be made a global flag. 

I would also suggest using the more common `verbose` naming. Before reading the help text I assumed this controls Constellation debug vs release images.